bitrepository-core/src/main/java/org/bitrepository/common/utils/TimeUtils.java

```diff
@@ -123,6 +132,90 @@ public static String millisecondsToHuman(long ms) {
         return sb.toString();
     }
 
+    /**
+     * Formats a non-negative Duration to an approximate human-readable string like "1y 2m" or "3h 45m".
+     * The conversion uses approximate average values for the lengths of days, months and years.
+     *
+     * The duration must be non-negative and not longer than 4 382 910 hours (approximately 500 years)
```

Comment:
The validation is in the wrong place here. It should be with the max age setting and should be documented there (since in practice the value comes from there). I suggest we fix that separately after both BITMAG-992 and BITMAG-1142 have been merged.